### DISCUSSION TITLE
[2023-06-26] Library and Engine Team Update

### DISCUSSION THREAD

**Discussion by jdunkerley** [enso-org/enso#7127] [Weekly Demo Day]
- James: Bug fix for Value constructors. #7080 
- James: Code clean up and tag clean up as part of working through group labelling. #7114 
- Radek: Execution context work for writing to the database. #7072 
- Greg: Continued work on rounding within a table (both in-memory and in-database). #6988
- Jaroslav: Continuing work on resolving drop-downs widgets for extension methods. #7115 
- Jaroslav: Improved Enso integration with VS Code, IGV, etc. #7054 
- Jaroslav: Avoid dependency on BindingMaps for IR. #7055 
- Jaroslav: Moved construction of IR into runtime parser project. #7096 
- Jaroslav: Bug fix for using state in Vector.map. #7078 
- Hubert: Add retries to the GraalVM updater. #7079 
- Hubert: Don't instrument function bodies (fixes map perfromance). #7097 
- Hubert: Bug fix for Tail Call optimizations with warnings. #7116 
- Hubert: Modifying `executionContext/executionComplete` reply to exclude failed executions
- Pavel: Always call instance methods on Any. #7033
- Pavel: Bug fix so the repl doesn't throw an NPE on empty input. #7111 
- Pavel: Working to invoke Enso benchmarks via JMH. #7101 
- Dmitry: Working to add method call info for infix operators. #7090 
- Dmitry: Do not invalidate the caches when building the engine distribution. #7106

**Comment by radeusgd** [comment]
## Create Table

![image](https://github.com/enso-org/enso/assets/1436948/54dc7da7-a769-4a14-883c-6eacf98f0aa3)
![image](https://github.com/enso-org/enso/assets/1436948/b1f617d4-608a-4907-9389-96507ef2f5ac)
![image](https://github.com/enso-org/enso/assets/1436948/139cb1f4-9726-480e-9c07-0658abe5be1a)
![image](https://github.com/enso-org/enso/assets/1436948/1687f389-bd98-4d2c-8579-2db2f45f71c1)


## Upload Table

![image](https://github.com/enso-org/enso/assets/1436948/30e0a3d0-1abd-4498-9042-f9818ab52fc5)
![image](https://github.com/enso-org/enso/assets/1436948/af5266fb-c6d5-4193-8573-ff9cd73ec675)
![image](https://github.com/enso-org/enso/assets/1436948/f63133cb-caf3-44b4-99ee-fc28ded922cf)

**Comment by radeusgd** [comment]
### Updating an existing Table

Dry run - no rows uploaded yet:
![image](https://github.com/enso-org/enso/assets/1436948/6b28bda1-e58b-4913-8af9-84a8232c65f0)

After hitting play:
![image](https://github.com/enso-org/enso/assets/1436948/3a07ed58-bc69-4cf2-a89b-bc6330ee7071)

After modifying the inputs - back to dry run - existing (old) table contents are shown:
![image](https://github.com/enso-org/enso/assets/1436948/0f3e8589-7983-49c9-a931-484310779a70)

After re-running again - the rows are added again:
![image](https://github.com/enso-org/enso/assets/1436948/a4a05c3f-2959-4733-910b-35019a3f240d)
